README.md

```diff
@@ -9,3 +9,77 @@ This work has been accepted into [BIoT 2018](http://cse.stfx.ca/~blockchain2018/
 ## License
 
 SmartEdge is made available under a permissive license - please see [LICENSE](LICENSE) for details.
+
+## Introduction
+
+We assume that the compute node initiates the contract because it would be expensive for data node to put out the contract since non serious compute nodes may end up wasting the ether of data nodes. It is imagined that there is only one data node and one compute node for the time being
+
+## States of the contract
+The entire transaction is imagined as a finite state machine and the states are as follows:
+1. Unavailable
+2. Available
+3. Pending
+4. Computing
+5. Completed
+
+## Initial state and working
+- We start in the unavailable state. The transaction involves the following 
+  - jobUrl:-The job url that the data node sends to the compute node 
+  - outputUrl:-The url containing output that compute node sends to data node
+- All Ether amounts stored in Wei
+
+## Transition Events
+1. MadeAvailable() 
+2. BidAccepted(address counterparty, string url)
+3. JobRejected()
+4. JobAccepted(uint startTime) 
+5. JobCompleted(uint startTime, uint endTime, string url) 
+6. JobTimedOut() 
+7. ResultVerified() 
+8. ResultTimedOut() 
+9. ResultRejected()
+
+## Modifiers
```

Comment:
In general the Modifiers and Functions sections are good, but some of the sentences need to be fixed or made more precise because they are incorrect.

For example, onlyBy() checks if a function is being called by the _specified_ address, not necessarily the owner in all cases.

Another example, resultVerified() is called by the counterparty, not the owner.